packages/web/components/transactions/transaction-content.tsx

```diff
@@ -1,83 +1,104 @@
 import { FormattedTransaction } from "@osmosis-labs/server";
+import { observer } from "mobx-react-lite";
 
+import { Spinner } from "~/components/loaders";
+import { NoTransactionsSplash } from "~/components/transactions/no-transactions-splash";
 import { TransactionRow } from "~/components/transactions/transaction-row";
 import {
   formatDate,
   groupTransactionsByDate,
 } from "~/components/transactions/transaction-utils";
 import { Button } from "~/components/ui/button";
 
-export const TransactionContent = ({
-  setSelectedTransaction,
-  transactions,
-  setOpen,
-  open,
-}: {
-  setSelectedTransaction: (selectedTransaction: FormattedTransaction) => void;
-  transactions: FormattedTransaction[];
-  setOpen: (open: boolean) => void;
-  open: boolean;
-}) => {
-  // TODO - add loading state
-  if (!transactions) return null;
-
-  const groupedTransactions = groupTransactionsByDate(transactions);
-
-  return (
-    <div className="flex w-full flex-col">
-      <div className="flex w-full justify-between pt-8 pb-4">
-        <h1 className="text-h3 font-h3">Transactions</h1>
-        <div className="flex gap-3">
-          <Button variant="secondary" size="md">
-            Explorer &#x2197;
-          </Button>
-          <Button variant="secondary" size="md">
-            Tax Reports &#x2197;
-          </Button>
+export const TransactionContent = observer(
+  ({
+    setSelectedTransaction,
+    transactions = [],
+    setOpen,
+    open,
+    address,
+    isLoading,
+    isWalletConnected,
+  }: {
+    setSelectedTransaction: (selectedTransaction: FormattedTransaction) => void;
+    transactions?: FormattedTransaction[];
```

Comment:
Heads up, we'll probably eventually want to make this a generic type that contains the min data needed to display transaction content, rather than a type defined by the Numia API response.

Alternatively, we could do this processing in the tRPC procedure and define the base tx content type on the server. Either works once we want to incorporate pending txs.